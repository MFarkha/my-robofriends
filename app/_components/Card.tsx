import React from "react";
import { IRobot } from "../page";

type CardProps = {
  robot: IRobot;
  key: number;
};

const Card = ({ robot }: CardProps) => {
  const { name, email, id } = robot;
  return (
    <div className="bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5">
      <img
        alt="robots"
        src={`https://robohash.org/${id}.png?size=200x200`}
        width="200"
        height="200"
      ></img>
      <div>
        <h2 className="f5">{name}</h2>
        <p>{email}</p>
      </div>
    </div>
  );
};

export default Card;
