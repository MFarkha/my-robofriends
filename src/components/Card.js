import React from 'react'

const Card = ({name, email, id}) => {
    return (
        <div className='tc bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5'>
            <img alt='robots' src={`https://robohash.org/${id}.png?size=200x200`} width='200' height='200'></img>
            <div>
            <h2 className='f5'>{name}</h2>
            <p>{email}</p>
            </div>
        </div>
    );
}

export default Card;