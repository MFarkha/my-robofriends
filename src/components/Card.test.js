import Card from './Card';
import React from 'react';
import { shallow } from 'enzyme';

it('expects to render the Card component', () => {
    const { mockId, mockName, mockEmail } = {
        id: 1,
        name: "John Snow",
        email: "john@example.com"
    }
    const wrap = shallow(<Card name={mockName} email={mockEmail} id={mockId}/>);
    expect(wrap).toMatchSnapshot();
})
