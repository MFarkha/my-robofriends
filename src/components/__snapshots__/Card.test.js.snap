// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expects to render the Card component 1`] = `
<div
  className="bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5"
>
  <img
    alt="robots"
    height="200"
    src="https://robohash.org/undefined.png?size=200x200"
    width="200"
  />
  <div>
    <h2
      className="f5"
    />
    <p />
  </div>
</div>
`;
